### PR TITLE
PolymerTokenization for GT4SD

### PR DESCRIPTION
**Pull request by jannisborn**
- Add PolymerTokenizer
- fix two lines for compatibility with recent transformers
- blackening file

### CODE REVIEW

File: terminator/tokenization.py [hunk 1]
```diff
@@ -166,6 +167,32 @@ def tokenize(self, text: str) -> List[str]:
         return self.tokenizer.tokenize(text)
 
 
+class PolymerTokenizer:
```
Comment (drugilsberg) [inline terminator/tokenization.py]: May I suggest to call this in a different way? Probably `PolymerGraphTokenizer` is more appropriate in this case.

File: terminator/tokenization.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 logging.basicConfig(stream=sys.stdout, level=logging.INFO)
 
 SMILES_TOKENIZER_PATTERN = r"(\%\([0-9]{3}\)|\[[^\]]+]|Br?|Cl?|N|O|S|P|F|I|b|c|n|o|s|p|\||\(|\)|\.|=|#|-|\+|\\|\/|:|~|@|\?|>>?|\*|\$|\%[0-9]{2}|[0-9])"
+POLYMER_TOKENIZER_PATTERN = r"(\%\([0-9]{3}\)|\[[^\]]+]|Br?|Cl?|N|O|S|P|F|I|b|c|n|o|s|p|A|B|C|D|E|R|Q|Z|;|<|>|\||\(|\)|\.|=|#|-|\+|\\|\/|:|~|@|\?|>>?|\*|\$|\%[0-9]{2}|[0-9])"
```
Comment (drugilsberg) [inline terminator/tokenization.py]:
```suggestion
POLYMER_GRAPH_TOKENIZER_PATTERN = r"(\%\([0-9]{3}\)|\[[^\]]+]|Br?|Cl?|N|O|S|P|F|I|b|c|n|o|s|p|A|B|C|D|E|R|Q|Z|;|<|>|\||\(|\)|\.|=|#|-|\+|\\|\/|:|~|@|\?|>>?|\*|\$|\%[0-9]{2}|[0-9])"
```

File: terminator/trainer.py [hunk 3]
```diff
@@ -661,7 +661,7 @@ def log(self, logs: Dict[str, float], iterator: Optional[tqdm] = None) -> None:
                 self.min_loss = logs["loss"]
                 self.save_model(output_dir)
                 # Save optimizer and scheduler
-                if self.is_world_master():
+                if self.is_world_process_zero():
```
Comment (drugilsberg) [inline terminator/trainer.py:664]: 🎩

File: terminator/tokenization.py (hunk 2 above)
Comment (jannisborn) [reply terminator/tokenization.py]: Yes, totally agreed, I was quite unsure about the names, I like your suggestion. Maybe we can also ask Nate, maybe he has a name in mind already for this

**Comment by jannisborn** [comment]
Changed the name of the tokenizer class

**Pull request merged**